forch/faucet_state_collector.py

```diff
@@ -1314,16 +1319,19 @@ def _get_access_switch(self, mac):
 
     def update_radius_result(self, mac, access, segment=None, role=None):
         """Update RADIUS result information for learned host"""
+        host_radius = {}
+        host_radius[MAC_RADIUS_ACCESS] = access
+        host_radius[MAC_RADIUS_SEGMENT] = segment
+        host_radius[MAC_RADIUS_ROLE] = role
+        self.radius_results[mac] = host_radius
```

Comment:
under what conditions is radius_results[mac} invalidated?  How are stale entries removed?

Reply:
So, authenticator drives the auth state of the port. If it gets deauthenticated ,radius result would have none in all of its entries and the status would be deauthenticated, which is accurate.